src/debug-generator/debug-generator.c

```diff
@@ -84,14 +95,11 @@ static int parse_breakpoint_from_string(const char *s, uint32_t *ret_breakpoints
 
         /* Empty value? set default breakpoint */
         if (isempty(s)) {
-                if (in_initrd()) {
-                        FOREACH_ARRAY(i, breakpoint_info_table, ELEMENTSOF(breakpoint_info_table))
-                                if (i->validity & BREAKPOINT_DEFAULT) {
-                                        breakpoints |= 1 << i->type;
-                                        break;
-                                }
-                } else
-                        log_warning("No default breakpoint defined on the host, ignoring breakpoint request from kernel command line.");
+                FOREACH_ARRAY(i, breakpoint_info_table, ELEMENTSOF(breakpoint_info_table))
+                        if (FLAGS_SET(i->validity, BREAKPOINT_DEFAULT) && breakpoint_applies(i, LOG_WARNING)) {
+                                breakpoints |= 1 << i->type;
+                                break;
+                        }
```

Comment:
Well, I doubt we should log here at all. The user expects us to decide a default value, during which there's no need to log anything.

OTOH, if we lack a default for the scope we should still log loudly, which is removed here however...